internal/globaldefault/default.go

```diff
@@ -0,0 +1,190 @@
+	"github.com/ActiveState/cli/internal/subshell/sscommon"
+	"github.com/ActiveState/cli/pkg/platform/runtime"
+)
+
+const shimDenoter = "!DO NOT EDIT! State Tool Shim !DO NOT EDIT!"
+
+type DefaultConfigurer interface {
+	Set(key string, value interface{})
+}
+
+func BinDir() string {
+	return filepath.Join(config.CachePath(), "bin")
+}
+
+func Prepare(subshell subshell.SubShell) error {
+	logging.Debug("Preparing globaldefault")
+	binDir := BinDir()
+
+	// Don't run prepare if we're already on PATH
+	path := strings.Split(os.Getenv("PATH"), string(os.PathListSeparator))
+	for _, p := range path {
+		if p == binDir {
+			return nil
+		}
+	}
+
+	if fail := fileutils.MkdirUnlessExists(binDir); fail != nil {
+		return locale.WrapError(fail.ToError(), "err_prepare_bin_dir", "Could not create bin directory.")
+	}
+
+	envUpdates := map[string]string{
+		"PATH": binDir,
+	}
+
+	if fail := subshell.WriteUserEnv(envUpdates, sscommon.Default, true); fail != nil {
+		return locale.WrapError(fail.ToError(), "err_prepare_update_env", "Could not write to user environment.")
+	}
+
+	return nil
+}
+
+// SetupDefaultActivation sets symlinks in the global bin directory to the currently activated runtime
+func SetupDefaultActivation(subshell subshell.SubShell, cfg DefaultConfigurer, runtime *runtime.Runtime, projectPath string) error {
+	logging.Debug("Setting up globaldefault")
+	if err := Prepare(subshell); err != nil {
+		return locale.WrapError(err, "err_globaldefault_prepare", "Could not prepare environment.")
+	}
+
+	env, fail := runtime.Env()
+	if fail != nil {
+		return errs.Wrap(fail, "Could not get runtime env")
+	}
+
+	envMap, err := env.GetEnv(false, "")
+	if err != nil {
+		return errs.Wrap(err, "Could not get env")
+	}
+
+	// roll back old symlinks
+	if err := cleanup(); err != nil {
+		return locale.WrapError(err, "err_rollback_shim", "Could not clean up previous default installation.")
+	}
+
+	// Retrieve artifact binary directory
+	var bins []string
+	if p, ok := envMap["PATH"]; ok {
+		bins = strings.Split(p, string(os.PathListSeparator))
+	}
+
+	exes, err := exeutils.Executables(bins)
+	if err != nil {
+		return locale.WrapError(err, "err_symlink_exes", "Could not detect executables")
+	}
+
+	// Remove duplicate executables as per PATH and PATHEXT
+	exes, err = exeutils.UniqueExes(exes, os.Getenv("PATHEXT"))
+	if err != nil {
+		return locale.WrapError(err, "err_unique_exes", "Could not detect unique executables, make sure your PATH and PATHEXT environment variables are properly configured.")
+	}
+
+	if err := createShims(exes); err != nil {
+		return locale.WrapError(err, "err_createshims", "Could not create shim files to set up the default runtime environment.")
+	}
+
+	cfg.Set(constants.GlobalDefaultPrefname, projectPath)
+
+	return nil
+}
+
+// cleanup removes all shims in the global binary directory that target a previous default project
+// If the target of a shim does not exist anymore, the shim is also removed.
```

Comment:
This comment should be removed.  The function now just removes all files in the directory.